.github/workflows/build-snap.yml

```diff
@@ -81,16 +81,25 @@ jobs:
         if: always()
         run: |
           mkdir -p logs
-          models=$(juju models --format json | jq -r .models[].name)
-          for model in $models; do name=$(echo $model | cut -d/ -f2); juju status -m $model -o logs/$name.yaml; cat logs/$name.yaml; done
-          for model in $models; do name=$(echo $model | cut -d/ -f2); juju debug-log -m $model --replay > logs/$name-debug-log.txt; done
           cp -rf $HOME/snap/openstack/common/logs/*.log logs/
+          models=$(juju models --format json | jq -r .models[].name)
+          for model in $models;
+          do 
+            name=$(echo $model | cut -d/ -f2);
+            juju status -m $model -o logs/$name.yaml;
+            cat logs/$name.yaml;
+            juju debug-log -m $model --replay > logs/$name-debug-log.txt;
+            for pod in $(sudo microk8s.kubectl get pods -n $name -o=jsonpath='{.items[*].metadata.name}');
+            do
+              sudo microk8s.kubectl logs --ignore-errors -n $name --all-containers $pod > logs/$pod.log;
```

Comment:
We should add a log and handle when the pod log can't be collected. (Maybe retry also with a -p ?)

Reply:
handled in subsequent push. In the failed case the pod is getting terminated and so the error. Retry may not be required.